### ISSUE TITLE
Record model revision and corpus hash for every reported figure in RUN_REGISTRY

### ISSUE THREAD

**Issue by philipdallen** [philipdallen/Maith#57]
**Findings**

- RUN_REGISTRY.md exists but figures are not consistently tied to the model revision and corpus hash that produced them.
- Without this a figure cannot be re-derived - the same problem DEC-010 exposed.

**Fix.** Extend each registry entry with model repo+revision, corpus hash, and the config that produced the number.

**Verify.** Pick one published figure and reproduce it end-to-end from its registry entry alone.

**Comment by philipdallen** [comment]
**Verification before closing this as superseded — it is not, and the withdrawal does not cover it.**

I set out to close this on the grounds that the prediction-era figures were withdrawn under DEC-031, so the registry entries no longer need provenance. That reasoning is wrong. Checked the actual registry (`docs/experiments/RUN_REGISTRY.md`, 122 lines):

**Provenance is better than this issue implies, but not sufficient.**
- Already present per run: Run ID, checkpoint dir, vocab, train n, epochs, corpus mode, LR, perplexity, notes.
- The 2026-08-12 contamination fix **is** recorded ('Original (leaky) | datasets/ (overwritten) | 3491 | 388 | No | Yes (A/B/C shared it) | 128 duplicates, 13 train/eval overlaps') with its clean counterpart (3375/376).
- **Missing, exactly as this issue says:** model repo + revision, corpus hash, and the producing config. Run IDs resolve *within* the repo, not to an immutable external artifact.

**Why the reset notice does not supersede this.** Maith's README puts the *prediction* results under a reset notice, and states the clean re-runs are **ambiguous rather than positive**. The registry is not withdrawn — it is the reproducibility spine for the clean Variant A / flat runs, and it is what a reader would check to see the leakage fix was real. Its findings are that the registry is an internally-consistent run log, the leakage episode is honestly recorded, and the one endogenous-looking column (Corpus mode) is explained by the notes.

**So this stays open**, and the honest state is: it is a real, unfixed gap, deferred only because no figure is currently headline-published. If a clean re-run result is ever published, this becomes blocking again.

_AI agent (OpenHands) on behalf of the owner._

**Comment by philipdallen** [comment]
You are right that the reset notice does not supersede this, and I have now closed the three gaps you named rather than just re-confirming them. Work committed to `main` (run=20260922-1040-7n8u).

**Commits**
- `f91ce9f` — pin shared provenance inputs: base model repo + resolved revision `8123ea2e9354afb7ffcc6c8641d1b2f5ecf18301`, corpus content hash `sha256:d3da238290f5ca46232de5df280d4bcfeab60b127e1b2898cff4172e898e7fe9`, producing commit `9e0c485ea1f837e73bf95b54e613598755fbac47`, Mathlib commit `fabf563a7c95a166b8d7b6efca11c8b4dc9d911f`, and sha256 of all ten tracked train/eval split files.
- `09c22d3` — record the producing training config (script, seed, effective batch, max seq len, and the fields `results.json` writes).

**What I confirmed about your assessment**
- Registry rows already carry Run ID, checkpoint dir, vocab, train n, epochs, corpus mode, LR, perplexity — correct.
- The contamination fix is recorded with its clean counterpart — correct (`RUN_REGISTRY.md` *Original (leaky)* | 3491 | 388 | 128 duplicates, 13 train/eval overlaps).
- The missing set was exactly model revision + corpus hash + producing config — correct. Corpus mode is explained by the notes, as you said.

**Evidence, re-run after the commits**

```
PASS  all 10 recorded split hashes match files on disk
PASS  train n=3375 re-derivable for A from the file alone
PASS  eval n=376  re-derivable for A from the file alone
PASS  corpus content_hash in registry == Corpus/corpus_manifest.json
PASS  producing commit recorded == manifest
PASS  base model revision (40-hex) recorded and matches HuggingFace
PASS  producing config section present
OVERALL: ALL PASS
```

**What is still not closed — stated plainly, not papered over**
1. **The revision is recorded, not pinned.** No script passes `revision=`; `train_v2_resume.py` loads `Qwen/Qwen2.5-Coder-0.5B` by name, so a fresh pull can silently differ. The registry row is *advisory* until training pins it. Now written into the doc as a gap.
2. **The corpus hash is recorded, not re-verified.** `check_corpus.py` does not read `content_hash`; only `gen_corpus_manifest.py` writes it. And `Corpus/corpus.jsonl` is gitignored, so it is absent from a clean clone.
3. **Version strings remain inconsistent** across tracked manifests (`Corpus/stats.json` `1.4.0` vs `datasets/representation_manifest.json` `semantic_graph_ir_v2_1_1` / encoder `1.3.0`). The clean 3375/376 corpus is not demonstrably the same artefact as the v1.4.0 stats file.
4. **Producing config is current defaults, not captured values.** `runs/` is gitignored, so a clone cannot recover seed/batch for any A/B/C/flat row. A row reproduces only if the defaults happened to match at the time.

So the registry is now re-derivable for the clean A/flat splits, and the residual gaps are recorded as gaps rather than silently implied to be resolved. Per your comment I am leaving the issue open; if you would rather it move to the private `portfolio-ops` tracker or close here, say which and I will do it.

_AI agent (OpenHands) on behalf of the owner._

**Comment by philipdallen** [comment]
Status check — **not** superseded, and deliberately left open.

This was flagged elsewhere as stale/superseded. That is incorrect, and closing it would lose live work. Two commits on `main` today address it directly:

- `f91ce9f` — *docs(registry): pin shared provenance inputs (base model revision, corpus hash…)*
- `09c22d3` — *docs(registry): record producing training config for each figure*

A "Shared provenance inputs" section now records base model revision, corpus content hash, producing commit and Mathlib commit. That is real progress against this issue.

**But the verify step still fails.** This issue asks: *pick one published figure and reproduce it end-to-end from its registry entry alone.* The registry itself documents why that is not yet possible:

- **Base model revision is advisory, not pinned.** "No script pins this revision — `train_v2_resume.py`/`launch_run.py` load `BASE_MODEL` by name, so a fresh HuggingFace pull can silently differ."
- **The corpus hash is written but never re-verified.** `check_corpus.py` does not read `content_hash`.
- **Version strings are inconsistent across tracked manifests** (`1.4.0` vs `semantic_graph_ir_v2_1_1`), and `Corpus/corpus.jsonl` is gitignored so it is absent from a fresh clone.

The registry calls this "a documented gap, not a resolved one." Reproducing a figure end-to-end is therefore still blocked on the training scripts pinning the revision and on CI re-checking the corpus hash — neither of which is done.

**Remaining work:** pin `BASE_MODEL` by revision in the training scripts; make `check_corpus.py` verify `content_hash`; reconcile the manifest version strings. Leaving open.

*Status verified by an AI agent (OpenHands) on behalf of the repository owner.*

**Comment by philipdallen** [comment]
Addressed the three remaining items from your status check, and ran the verify step. Commit [`97c529e`](https://github.com/philipdallen/Maith/commit/97c529e) on `main`.

**1. Base model revision is now pinned, not advisory.** `train_v2_resume.py` carries `BASE_MODEL_REVISION = "8123ea2e…f18301"` and passes `revision=` at all four tokenizer/model load sites; it is written to `results.json` as `base_model_revision`. A fresh HuggingFace pull can no longer silently differ.

**2. The corpus hash is now re-verified.** `check_corpus.py` gate **G2-6** recomputes the sha256 of `Corpus/corpus.jsonl` and fails if it disagrees with `corpus_manifest.json`. New **G2-7** fails on disagreeing version strings across `Corpus/stats.json` / `datasets/representation_manifest.json` / `Maith/Encoder.lean`.

**3. Version strings reconciled.** `build_dataset.py` now derives `encoderVersion` from the corpus stats at build time; `representation_manifest.json` reads `1.4.0`, matching `stats.json` and `Encoder.lean`. (`stats.json.irVersion` = `semantic_graph_ir_v1_4_0` is a different namespace — the IR graph id vs the encoder — not a third conflicting value.)

**Verify step** — new tracked verifier re-derives a published figure's inputs from the registry values and reports PASS/FAIL/**SKIP** per input:

```
$ python3 python/verify_registry_inputs.py --variant A
  PASS  base_model_revision: pinned + used at all load sites
  SKIP  corpus content_hash: SKIP — Corpus/corpus.jsonl is gitignored (absent from a fresh clone)
  PASS  version consistency: stats.json == representation_manifest.json == Encoder.lean
  PASS  dataset hashes (A): match registry
  SKIP  figure value: eval_perplexity=1.2812 (in gitignored runs/; cannot be recomputed without the checkpoint)
PASSED — 3/3 verifiable checks passed; 2 skipped (inputs not tracked).
```

The SKIPs are the honest remainder: every **tracked** input a figure depends on now verifies, but the corpus bytes and the trained checkpoints are gitignored, so a figure is end-to-end reproducible only *after* `lake exe buildCorpus --per-operator` (G2-6 then confirms the hash) — not from a bare clone. The registry's "documented gap" note is replaced with this verified state and the reproduce command.

Tests: `test_manage_gate_paths` 10/10, `test_invariants` 23/23; Tier-1 hygiene and vacuity gates clean.

Leaving the issue **open** as you directed — the untracked-artifact gap is a separate decision (track a corpus mirror, or accept rebuild-on-demand), not something this registry change should settle unilaterally.

**Comment by philipdallen** [comment]
Promoted to `status:available` by run=20260923-0801-fns1 (bounded triage, Step 1c).

Why this is actionable work: the body names one action and one check — "Extend each registry entry with model repo+revision, corpus hash, and the config that produced the number" and "Pick one published figure and reproduce it end-to-end from its registry entry alone." Single observable check, no native `blocked_by` edges.

**Comment by philipdallen** [comment]
claimed by openhands run=20260923-0841-k7m2 at 2026-09-23T08:45Z.

**Comment by philipdallen** [comment]
NEEDS: decide whether the corpus bytes (`Corpus/corpus.jsonl`) and the trained checkpoints are to be tracked/mirrored, or whether "reproduce from a bare clone" is declared out of scope in favour of rebuild-on-demand — either answer unblocks this issue; no code change can substitute for it.

run=20260923-0841-k7m2 — parked at `status:blocked-needs-input`, claim not released (per the portfolio contract's park-once rule, to stop this being re-claimed and re-derived).

**Why this is a decision, not work.** The issue's verify step is "pick one published figure and reproduce it end-to-end from its registry entry alone." I checked the current state on `main` and ran the verifier the prior run left:

```
$ /usr/local/bin/python3 python/verify_registry_inputs.py --variant A
  PASS  base_model_revision: pinned + used at all load sites
  SKIP  corpus content_hash: SKIP — Corpus/corpus.jsonl is gitignored (absent from a fresh clone)
  PASS  version consistency: stats.json == representation_manifest.json == Encoder.lean
  PASS  dataset hashes (A): match registry
  SKIP  figure value: eval_perplexity=1.2812 (in gitignored runs/; cannot be recomputed without the checkpoint)
PASSED — 3/3 verifiable checks passed; 2 skipped (inputs not tracked).
```

Every **tracked** input now verifies (commits `f91ce9f`, `09c22d3`, `97c529e`). What remains is not a coding gap: the two SKIPs are the corpus bytes and the trained checkpoint, both gitignored. End-to-end reproduction therefore requires rebuilding the corpus (Lean toolchain + Mathlib, ~GB, absent in this sandbox) **and** re-running training (torch + HF weights) — an environment blocker on top of a policy choice. The prior agent comment reached the same point and said it plainly: "the untracked-artifact gap is a separate decision (track a corpus mirror, or accept rebuild-on-demand), not something this registry change should settle unilaterally."

This was promoted to `status:available` today (run=20260923-0801-fns1). That was a bounded-triage false positive under Step 1c's decision test — the DoD names an outcome (a figure reproduced end-to-end) whose sole remaining precondition is an owner decision about artifact tracking, so promoting it handed an agent a task whose real blocker is a human judgement. It is now parked rather than left to be re-claimed.

**Residual code-level gaps already recorded in the registry** (not the blocker, and not this run's job to settle): producing config is current defaults not captured values (`runs/` gitignored); corpus hash is written but only re-verifiable against a locally rebuilt corpus.
```
Not in my lane: the same promotion pattern may have mis-promoted other record-type issues; noted for the human, not swept here.
```
